zkvm/docs/zkvm-spec.md

```diff
@@ -1390,49 +1412,90 @@ Fails if:
 
 #### input
 
-_input_ **input** → _contract_
+_prevoutput_ **input** → _contract_
 
-1. Pops a [data](#data-type) `input` representing the [input structure](#input-structure) from the stack.
-2. Constructs a [contract](#contract-type) based on the `input` data and pushes it to the stack.
-3. For each decoded [value](#value-type), quantity variable is allocated first, flavor second.
-4. Adds [input entry](#input-entry) to the [transaction log](#transaction-log).
-5. Sets the [VM uniqueness flag](#vm-state) to `true`.
+1. Pops a [data](#data-type) `prevoutput` representing the [unspent output structure](#output-structure) from the stack.
+2. Constructs a [contract](#contract-type) based on the `prevoutput` data and pushes it to the stack.
+3. Adds [input entry](#input-entry) to the [transaction log](#transaction-log).
+4. Sets the [VM’s last anchor](#vm-state) to the ratcheted [contract ID](#contract-id):
+    ```
+    T = Transcript("ZkVM.ratchet-anchor")
+    T.commit("old", contract_id)
+    new_anchor = T.challenge_bytes("new")
+    ```
+
+Fails if the `prevoutput` is not a [data type](#data-type) with exact encoding of an [output structure](#output-structure).
```

Comment:
I think this whole document needs to replace the way it uses "data type." Brought this up [in Slack](https://i10r.slack.com/archives/CCVPR72CR/p1549411686017000?thread_ts=1549408834.013200&cid=CCVPR72CR) originally:

> the way the spec uses the phrase "data type" is causing some cognitive dissonance. e.g. "scalars are encoded as 32-byte data types." those are _strings_ (or _blobs_ or _bytearrays_ or something). a "data type" should be a type, not data

Reply:
Let's revisit this in another PR. I agree that "data" is confusing, but "blob" and "bytearray" sound clunky and "string" sounds too much like "human text" (e.g. most modern languages have String type for UTF8 text). Motivation for "data" was as opposite to "non-data" linear types like "Values" and "Contracts". Those are not _data types_ because those are _things_. But i don't deny there's confusion alike to would-be one with String-as-bytes VS String-as-humantext.

Reply:
Sounds good. To get a little pedantic, in the phrase "data type," "type" works as a noun and "data" works as an adjective modifying it; so it's a kind of type, not a kind of data. (The phrase "type data" would be a kind of data, not a kind of type.) My objection is not to the word "data" per se, but to "type" used as a noun to describe data.

Reply:
Filed: 
https://github.com/interstellar/slingshot/issues